tests/python_api/last_batch_policy_testing.py

```diff
@@ -0,0 +1,178 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from amd.rocal.pipeline import Pipeline
+from amd.rocal.pipeline import pipeline_def
+from amd.rocal.plugin.pytorch import ROCALAudioIterator
+import amd.rocal.fn as fn
+import amd.rocal.types as types
+import random
+import os
+import sys
+import matplotlib.pyplot as plt
+import torch
+import numpy as np
+from parse_config import parse_args
+
+np.set_printoptions(threshold=1000, edgeitems=10000)
+seed = random.SystemRandom().randint(0, 2**32 - 1)
+
+test_case_augmentation_map = {
+    0: "last_batch_policy_FILL_last_batch_padded_True",
+}
+
+def plot_audio_wav(audio_tensor, idx):
+    # audio is expected as a tensor
+    audio_data = audio_tensor.detach().numpy()
+    audio_data = audio_data.flatten()
+    plt.plot(audio_data)
+    plt.savefig("OUTPUT_FOLDER/AUDIO_READER/" + str(idx) + ".png")
+    plt.close()
+
+def verify_output(output_list, rocal_data_path, roi_tensor, test_results, case_name, dimensions):
+    ref_path = f'{rocal_data_path}/GoldenOutputsTensor/reference_outputs_audio/{case_name}_output.bin'
+    data_array = np.fromfile(ref_path, dtype=np.float32)
+    audio_data = output_list[0].detach().numpy()
+    audio_data = audio_data.flatten()
+    roi_data = roi_tensor.detach().numpy()
+    buffer_size = roi_data[0] * roi_data[1]
+    matched_indices = 0
+    for i in range(roi_data[0]):
+        for j in range(roi_data[1]):
+            ref_val = data_array[i * roi_data[1] + j]
+            out_val = audio_data[i * dimensions[2] + j]
+            # ensuring that out_val is not exactly zero while ref_val is non-zero.
+            invalid_comparison = (out_val == 0.0) and (ref_val != 0.0)
+            #comparing the absolute difference between the output value (out_val) and the reference value (ref_val) with a tolerance threshold of 1e-20.
+            if not invalid_comparison and abs(out_val - ref_val) < 1e-20:
+                matched_indices += 1
+
+    # Print results
+    print(f"Results for {case_name}:")
+    if matched_indices == buffer_size and matched_indices != 0:
+        print("PASSED!")
+        test_results[case_name] = "PASSED"
+    else:
+        print("FAILED!")
+        test_results[case_name] = "FAILED"
+
+@pipeline_def(seed=seed)
+def audio_decoder_pipeline(path, file_list, downmix=False):
```

Comment:
Can we pass nshards as arg here and then use it in audio decoder func?

Reply:
Will do when making changes in Test suite